packages/wrangler/src/pages/dev.tsx

```diff
@@ -307,8 +310,44 @@ export const Handler = async ({
 		);
 	}
 
+	let entrypoint = scriptPath;
+
+	if (directory) {
+		const routesJSONPath = join(directory, "_routes.json");
+
+		if (existsSync(routesJSONPath)) {
+			const routesJSONContents = readFileSync(routesJSONPath, "utf-8");
+
+			try {
+				validateRoutes(JSON.parse(routesJSONContents), directory);
+			} catch (err) {
+				throw err;
+			}
+
+			entrypoint = join(tmpdir(), `${Math.random().toString(36).slice(2)}.js`);
+
+			await build({
```

Comment:
We should watch `_route.json` and re-build whenever it changes.

Reply:
✅